visualization/app/codeCharta/state/effects/coupleHeightAndColorMetric/coupleHeightAndColorMetric.effect.ts

```diff
@@ -0,0 +1,26 @@
+import { Inject, Injectable } from "@angular/core"
+import { Store } from "../../angular-redux/store"
+import { createEffect } from "../../angular-redux/effects/createEffect"
+import { combineLatest, map } from "rxjs"
+import { isHeightAndColorMetricCoupledSelector } from "../../store/appSettings/isHeightAndColorMetricCoupled/isHeightAndColorMetricCoupled.selector"
+
+import { setColorMetric } from "../../store/dynamicSettings/colorMetric/colorMetric.actions"
+import { heightMetricSelector } from "../../store/dynamicSettings/heightMetric/heightMetric.selector"
+import { colorMetricSelector } from "../../store/dynamicSettings/colorMetric/colorMetric.selector"
+
+@Injectable()
+export class CoupleHeightAndColorMetricEffect {
+	constructor(@Inject(Store) private store: Store) {}
+
+	coupleHeightAndColorMetric$ = createEffect(() =>
+		combineLatest([
+			this.store.select(heightMetricSelector),
+			this.store.select(isHeightAndColorMetricCoupledSelector),
+			this.store.select(colorMetricSelector)
+		]).pipe(
+			map(([heightMetric, isCouplingEnabled, colorMetric]) => {
+				return isCouplingEnabled ? setColorMetric(heightMetric) : setColorMetric(colorMetric)
+			})
+		)
+	)
```

Comment:
@shaman-apprentice Do you have a suggestion on how can I approve this effect? It also dispatches the color metric when the coupling is false, but it's not necessary (I don't know any better). It should only dispatch the color metric when the coupling is enabled and the height metric is also changed. So maybe I need a kind of filter when I combine these selectors?